sympy/functions/elementary/integers.py

```diff
@@ -587,6 +587,51 @@ def _value_one_or_more(self, other):
             if other.is_integer and other.is_positive:
                 return S.true
 
+    def _eval_as_leading_term(self, x, logx=None, cdir=0):
+        from sympy.calculus.accumulationbounds import AccumBounds
+        arg = self.args[0]
+        arg0 = arg.subs(x, 0)
+        r = self.subs(x, 0)
+
+        if arg0.is_finite:
+            if r.is_zero:
+                if cdir == 0:
+                    ndirl = arg.dir(x, cdir=-1)
+                    ndir = arg.dir(x, cdir=1)
+                    if ndir != ndirl:
+                        raise ValueError("Two sided limit of %s around 0"
+                                            "does not exist" % self)
```

Comment:
Two sided limit should be handled by the limit code. Here, and in most other leading term methods, `cdir=0` should be handled in the same way as `cdir=1` (by setting `cdir = 1`). There is no need to compute `ndir1`.

Reply:
Yeah I had given a thought on this but then I saw few functions related to `frac` (which I guess would be the `floor` function)  . And such a block forms a core part of `floor.as_leading_term()`. 
```
    def _eval_as_leading_term(self, x, logx=None, cdir=0):
        from sympy.calculus.accumulationbounds import AccumBounds
        arg = self.args[0]
        arg0 = arg.subs(x, 0)
        r = self.subs(x, 0)
        if arg0 is S.NaN or isinstance(arg0, AccumBounds):
            arg0 = arg.limit(x, 0, dir='-' if re(cdir).is_negative else '+')
            r = floor(arg0)
        if arg0.is_finite:
            if arg0 == r:
                if cdir == 0:
                    ndirl = arg.dir(x, cdir=-1)
                    ndir = arg.dir(x, cdir=1)
                    if ndir != ndirl:
                        raise ValueError("Two sided limit of %s around 0"
                                    "does not exist" % self)
                else:
                    ndir = arg.dir(x, cdir=cdir)
                return r - 1 if ndir.is_negative else r
            else:
                return r
        return arg.as_leading_term(x, logx=logx, cdir=cdir)
```
But yeah I'll try the other way round and maybe refactor floor and ceiling functions too along with the frac function.